plugins/bundle/plugin.go

```diff
@@ -509,7 +516,11 @@ func (p *Plugin) process(ctx context.Context, name string, u download.Update) {
 			return
 		}
 
-		if u.Bundle.Type() == bundle.SnapshotBundleType && p.persistBundle(name) {
+		p.cfgMtx.RLock()
+		src := p.config.Bundles[name]
+		p.cfgMtx.RUnlock()
+
```

Comment:
If the write lock on the config was acquired at this point, won't it affect `persistBundle`?

Reply:
I couldn't see a concern with `persistBundle` as src is read to a new variable and passed. However if we acquire the write lock for the whole method of `Reconfigure` as in previous comment, then this read lock will be a problem. 
Not sure if I understood your concern here.